src/ralph/models/xapi/__init__.py

```diff
@@ -2,10 +2,18 @@
 
 # flake8: noqa
 
+from .lms.statements import (
+    LMSAccessedFile,
+    LMSAccessedPage,
+    LMSDownloadedFile,
+    LMSDownloadedVideo,
+    LMSRegisteredCourse,
+    LMSUnregisteredCourse,
+    LMSUploadedFile,
+)
 from .navigation.statements import PageTerminated, PageViewed
 from .video.statements import (
     VideoCompleted,
-    VideoDownloaded,
```

Comment:
Isn't `VideoDownloaded` just a special case of the `FileDownloaded` statement?
Do we intend to have `PDFDownloaded`, `ImageDownloaded`, etc. statements?

Reply:
Why not ? If we specify a template with a specific object ID. Do you think we should propose a first draft of the whole possible downloading activities ?

Reply:
Great Idea to propose some of the most common download activities) and probably upload/accessed too)

I was wondering whether we should be specific or generic regarding the downloaded statement:

- The generic approach (`FileDownloaded`) seems simple to adopt and maintain; however, it might lack details about the downloaded object.
- The specific approach (`VideoDownloaded`, `PDFDownloaded`, etc.), on the other hand, would let us create more detailed statements; however, this might result in a more complex, harder-to-follow profile.

I think keeping both, as it is now, is probably best)
However, in this case, we have two ways to describe a download activity e.g. `john downloaded video` and  `john downloaded file of type video`.
Maybe we could add an exclude rule for the `FileDownload`.`object`.`definition`.`extensions`.`type` to not accept any video types? And also specify in the template description that the generic download statement template should only be used if no specific download statement template is defined?

Reply:
I agree with you to keep both combinations.
If we keep the list of authorized file types in object extensions (withou `video`), then it respects the fact that the video downloading can not be modelized with this template